catalogue/webapp/components/WorkDetails/WorkDetails.js

```diff
@@ -262,28 +266,30 @@ const WorkDetails = ({
               <AudioPlayer audio={audio} />
             </Space>
           )}
-          {itemLinkState === 'useLibraryLink' && (
-            <Space
-              as="span"
-              h={{
-                size: 'm',
-                properties: ['margin-right'],
-              }}
-            >
-              <ButtonSolidLink
-                icon="eye"
-                text="View"
-                trackingEvent={{
-                  category: 'WorkDetails',
-                  action: 'follow view link',
-                  label: work.id,
+          {itemLinkState === 'useLibraryLink' &&
```

Comment:
What is this piece of logic trying to say?

Reply:
Sorry, this is horrible and shouldn't be here anyway. I've move the logic into the getItemLinkState function, which should be clearer